### PR TITLE
fix(web): guard against expired session at boot and on a timer

### PR DESCRIPTION
**Pull request by katspaugh**
> Cookie expires, Redux still says authenticated,
> components flood the gateway with credentialed 403s.
> One /me probe at boot — and a timer for the rest.

## What it solves

Resolves: when the JWT session cookie expires, the browser stops sending it but Redux's \`sessionExpiresAt\` still says the user is signed in. Components keep firing credentialed requests against the gateway, every one of which 403s.

## How this PR fixes it

New \`useSessionExpiryGuard\` hook, registered once in \`InitApp\`. After store hydration, when the persisted state says the user is signed in:

1. If \`sessionExpiresAt\` has already passed → clear auth + toast immediately, no \`/v1/auth/me\` round-trip.
2. Else fire **one** \`/v1/auth/me\` probe; on 403 clear auth + toast, on 200 keep state.
3. Schedule a timer to clear auth + toast at \`sessionExpiresAt\` so a session that expires mid-tab is cleared without further credentialed requests.

Skipped while \`useLogoutCallback\`/\`useOidcLoginCallback\` are processing — those hooks already own \`/me\` for those flows.

**Backend behaviour confirmed unchanged.** \`auth.controller.ts:134-138\` already sets the cookie's \`Max-Age\` from the JWT \`exp\` claim, so the browser auto-deletes it on time. No backend change needed.

**Why this differs from the reverted #7807.** That PR added a global RTK Query 403 interceptor that conflated two unrelated 403s — \"cookie missing/invalid\" vs \"authenticated but not authorized\" (e.g. trying to delete a space you don't own). The backend \`AuthGuard\` returns identical 403s for both, so the interceptor false-positived on space deletion. This PR never inspects component-route 403s; only \`/v1/auth/me\` is consulted, where 403 unambiguously means \"no valid session.\"

## How to test it

1. Sign in to Spaces.
2. In DevTools → Application → Cookies, delete the \`access_token\` cookie (simulates expiry).
3. Reload the page. Expected: a single \`/v1/auth/me\` request returns 403, the \"Session expired\" toast appears, and no further credentialed requests fire (Network tab should not show a flood of 403s on \`/v1/spaces\`, \`/v1/users\`, etc.).
4. Negative test: sign in, leave the cookie alone, reload. Expected: a single \`/v1/auth/me\` returns 200, no toast, normal session continues.
5. Permission denial sanity check: as a non-owner, attempt an action that should 403 server-side. Expected: that action's 403 is **not** treated as session expiry — no toast, no logout.

## Affected flows

- App boot when a persisted SIWE/OIDC session is present.
- Mid-session expiry while the tab stays open (timer path).
- OIDC login callback and SIWE logout — guard intentionally yields to those.

## Blast radius

- New hook in \`apps/web/src/services/sessionExpiry/\`.
- One call site: \`apps/web/src/pages/_app.tsx\` (\`InitApp\`).
- Reads \`state.auth.sessionExpiresAt\`, \`selectIsStoreHydrated\`.
- Uses \`cgwApi.endpoints.authGetMeV1\` — same endpoint already used by \`reconcileAuth\` (\`useLogoutCallback\`, \`useOidcLoginCallback\`); RTK Query deduplicates concurrent calls.
- Coordinates with \`LOGGING_OUT_KEY\` (\`useLogoutCallback\`) and \`oidc_auth_pending\` (\`useOidcLoginCallback\`) sessionStorage flags.
- No mobile changes.
- No 403 interceptor: untouched downstream — does **not** affect Turnstile (returns 401, different code), space-deletion permission denials, or any other credentialed-route 403.

## Risks / not checked

- Did not run the mobile build (web-only fix).
- Did not exercise on a live staging session — verified in jsdom only.
- The post-sign-in \`sessionExpiresAt\` is still hardcoded to \`Date.now() + 24h\` in \`SignInButton\`. If the gateway's \`AUTH_VALIDITY_PERIOD_SECONDS\` is shorter, the timer fires too late; this PR catches that case via the boot-time \`/me\` probe but not for mid-session drift on the first session. Aligning the value belongs to a follow-up.

## Visual summary

\`\`\`mermaid
flowchart TD
  A[App boots / InitApp mounts] --> B{Store hydrated?}
  B -- no --> A
  B -- yes --> C{sessionExpiresAt}
  C -- null --> Z[no-op]
  C -- past --> D[setUnauthenticated + toast]
  C -- future --> E{LOGGING_OUT_KEY or oidc_auth_pending?}
  E -- yes --> Z
  E -- no --> F[GET /v1/auth/me]
  F -- 200 --> G[schedule timer for sessionExpiresAt]
  G -- timer fires --> D
  F -- 403 --> D
  F -- 5xx/network --> H[leave state untouched]
\`\`\`

## Checklist

- [ ] I've tested the branch on mobile 📱
- [x] I've documented how it affects the analytics (if at all) 📊 — no analytics impact
- [x] I've written a unit/e2e test for it (if applicable) 🧑‍💻
- [x] I've listed affected flows and blast radius, and named what I did not verify 🎯

---

## CLA signature

With the submission of this Pull Request, I confirm that I have read and agree to the terms of the [Contributor License Agreement](https://safe.global/cla).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by katspaugh** [comment]
@claude review

File: apps/web/src/services/sessionExpiry/useSessionExpiryGuard.ts [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+      return
+    }
+
+    // Don't compete with useLogoutCallback / useOidcLoginCallback — they already
+    // call /v1/auth/me as part of those flows. Both flags are set *synchronously
+    // before a full-page redirect* (see useLogout.ts / useOidcLogin.ts), so by
+    // the time we read them on the return load they're guaranteed to be present
+    // for the entire round-trip. Removal happens only after the callback's
+    // async /me resolves, after which a re-render with cleared flags lets us
+    // observe the post-flow state on the next effect pass.
```
[the remark this answers is not in the document]
Comment (katspaugh) [reply apps/web/src/services/sessionExpiry/useSessionExpiryGuard.ts]: Fixed in 9ca7ea6f4. The hook now arms the local-expiry timer up front, regardless of the OIDC/logout flag — only the `/me` probe is suppressed during those flows. So a callback that finishes silently (no auth-state dispatch, just the flag removed) still gets sessionExpiresAt enforced. Added a regression test [`useSessionExpiryGuard.test.tsx:175`](https://github.com/safe-global/safe-wallet-monorepo/blob/9ca7ea6f4/apps/web/src/services/sessionExpiry/__tests__/useSessionExpiryGuard.test.tsx#L175-L198) that asserts the timer fires after `LOGGING_OUT_KEY` is set.

File: apps/web/src/services/sessionExpiry/useSessionExpiryGuard.ts [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+    if (sessionExpiresAt <= Date.now()) {
+      expireNow()
+      return
+    }
+
+    const scheduleLocalExpiryTimer = () => {
+      if (cancelled) return
+      timer = setTimeout(expireNow, Math.max(0, sessionExpiresAt - Date.now()))
+    }
```
[the remark this answers is not in the document]
Comment (katspaugh) [reply apps/web/src/services/sessionExpiry/useSessionExpiryGuard.ts:107]: Fixed in 9ca7ea6f4. Timer is now armed via `setTimeout(expireNow, sessionExpiresAt - Date.now())` immediately after the pre-flight check, before the `/me` probe is dispatched. The probe runs in parallel and can override with an earlier 403; otherwise the timer is the floor on cleanup latency. Added a regression test [`useSessionExpiryGuard.test.tsx:222`](https://github.com/safe-global/safe-wallet-monorepo/blob/9ca7ea6f4/apps/web/src/services/sessionExpiry/__tests__/useSessionExpiryGuard.test.tsx#L222-L243) using a never-resolving probe to assert that auth is cleared when `sessionExpiresAt` passes regardless of probe state.

**Comment by katspaugh** [comment]
Addressed the @claude review items in 9ca7ea6f4:

- **Type safety on `isForbidden`** — fixed: now narrows to `error is FetchBaseQueryError` with a proper `'status' in error` check.
- **Redundant `typeof window !== 'undefined'`** — removed; the `useEffect` callback only runs client-side.
- **Edge case for slow probes** — added a new regression test that uses a never-resolving probe and asserts the local timer still expires the session at `sessionExpiresAt`.

Did not change two items from that review:

- **\"Race in cleanup\" (store timer ref before checking `cancelled`)** — the suggested change creates a timer just to immediately destroy it. The current order (check `cancelled` → only then call `setTimeout`) is correct: if cleanup has run, we must not schedule. Note that 9ca7ea6f4 also restructures this — the timer is now armed up front in the effect body (not inside an async callback), which sidesteps the concern entirely.
- **\"Memory leak risk with concurrent requests\" (AbortController)** — `dispatch(initiate())` returns an RTK Query subscription that we already release via `probe.unsubscribe()` in `.finally(...)`. RTK Query handles request lifecycle internally; an `AbortController` would be redundant. The `lastProcessedRef` also prevents the same `sessionExpiresAt` from spawning multiple probes.

**Comment by katspaugh** [comment]
Addressed item 1 in f4682ec78 — dropped the `(error as { status: unknown })` cast; with `typeof error === 'object' && error !== null && 'status' in error`, TypeScript narrows `error.status` directly. `type-check` is clean.

Items 2 and 3 dismissed:

- **Timer reference assignment** — `timer = setTimeout(expireNow, ...)` evaluates the right-hand side first (synchronous `setTimeout` call returning a handle), then assigns. The two-step `const timerId = setTimeout(...); timer = timerId` is identical in observable behavior; there is no window in which a pending timer exists without `timer` being assigned. And `clearTimeout(undefined)` is a documented no-op, so even the hypothetical "cleanup runs on undefined" case is harmless.
- **Unused `FetchBaseQueryError` import** — it *is* used, as the return type predicate of the type guard (`error is FetchBaseQueryError`). The narrowing flows to callers (`if (isForbidden(error)) expireNow()`), so the type isn't dead.

**Review by LucieFaire — APPROVED** [review]

**Comment by clovisdasilvaneto** [comment]
@katspaugh shouldn't we also intercept the backend request so if the user receives 403 we invalidate the session? In this way we not only relly in the browser data but also on what is returned to us in the BE

**Comment by katspaugh** [comment]
@clovisdasilvaneto that approach resulted in too many false positives, that's why it was reverted. In this PR, we don't rely only on browser data but make a special request to CGW to validate the token (/auth/me).
